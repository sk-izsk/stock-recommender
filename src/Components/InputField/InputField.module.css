.inputFieldContainer {
  display: flex;
  padding: 16px;
  flex: 1;
  justify-content: space-evenly;
}

.inputField {
  width: 100%;
}

input[type="text"] {
  padding: 10px;
  margin: 10px 5px;
  border: 2px solid cadetblue;
  border-radius: 10px;
  width: -webkit-fill-available;
  font-family: inherit;
  font-size: inherit;
}

input:focus,
textarea:focus,
select:focus {
  outline: none;
}

label {
  margin-left: 5px;
}
