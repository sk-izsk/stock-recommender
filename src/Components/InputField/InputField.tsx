import React, { ChangeEvent } from "react";
import styles from "./InputField.module.css";

interface Props {
  stockPriceValue: string;
  socialMediaCountValue: string;
  recommendationValue: string;
  stockPriceHandler: (e: ChangeEvent<HTMLInputElement>) => void;
  socialMediaCountHandler: (e: ChangeEvent<HTMLInputElement>) => void;
  stockRecommendationHandler: (e: ChangeEvent<HTMLInputElement>) => void;
}

const InputField: React.FC<Props> = ({
  stockPriceValue,
  socialMediaCountValue,
  recommendationValue,
  stockPriceHandler,
  socialMediaCountHandler,
  stockRecommendationHandler,
}) => {
  return (
    <div className={styles.inputFieldContainer}>
      <div className={styles.inputField}>
        <label>Find the stock price</label>
        <input
          type="text"
          placeholder="eg: APPL,02-01-21"
          name="stockPrice"
          value={stockPriceValue}
          onChange={stockPriceHandler}
        />
      </div>
      <div className={styles.inputField}>
        <label>Find social media count</label>
        <input
          type="text"
          placeholder="eg: APPL,Facebook"
          name="socialMediaCount"
          value={socialMediaCountValue}
          onChange={socialMediaCountHandler}
        />
      </div>
      <div className={styles.inputField}>
        <label>Find the stock recommendation</label>
        <input
          type="text"
          placeholder="eg: APPL"
          name="recommendation"
          value={recommendationValue}
          onChange={stockRecommendationHandler}
        />
      </div>
    </div>
  );
};

export { InputField };
