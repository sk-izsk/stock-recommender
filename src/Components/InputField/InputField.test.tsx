import { render } from "@testing-library/react";
import React from "react";
import { InputField } from "./InputField";

const mockProps = {
  stockPriceValue: "Mock Value",
  socialMediaCountValue: "Mock Counter",
  stockPriceHandler: jest.fn(),
  socialMediaCountHandler: jest.fn(),
};

test("App renders correctly", () => {
  const { asFragment } = render(<InputField {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
