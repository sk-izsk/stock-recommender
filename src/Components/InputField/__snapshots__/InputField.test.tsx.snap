// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="inputFieldContainer"
  >
    <div
      class="inputField"
    >
      <label>
        Find the stock price
      </label>
      <input
        name="stockPrice"
        placeholder="eg: APPL,02-01-21"
        type="text"
        value="Mock Value"
      />
    </div>
    <div
      class="inputField"
    >
      <label>
        Find social media count
      </label>
      <input
        name="socialMediaCount"
        placeholder="eg: APPL,facebook"
        type="text"
        value="Mock Counter"
      />
    </div>
  </div>
</DocumentFragment>
`;
